drv/update-api/src/lib.rs

```diff
@@ -128,6 +129,10 @@ impl From<UpdateError> for GwUpdateError {
             UpdateError::ImageBoardUnknown => Self::ImageBoardUnknown,
             UpdateError::TaskRestarted => Self::TaskRestarted,
             UpdateError::NotImplemented => Self::NotImplemented,
+
+            // TODO: Update gateway error to handle this then change the error to
+            // a proper enum
```

Comment:
I'm starting to really question this strategy. We won't actually know when all MGSs are updated with the latest errors and so if we send back something other than `Unknown(error_code)`, MGS may not be able to deserialize it.

It would be nice to have a way in gateway-messages, for errors in particular, to be able to check to see if they can deserialize, and if not, log the discriminant byte of the error in an `Unknown` value. This probably can't be automatic for all errors, but it seems like a way around the non-atomic update issue such that we wouldn't perpetually send `Unknown(code)` for each new error variant that wasn't there originally.

Curious for your thoughts @jgallagher @cbiffle @mkeeter